Dockerfile

```diff
@@ -0,0 +1,34 @@
+FROM gradle:8.10-jdk21-jammy AS base
+
+WORKDIR /app/
+
+COPY build.gradle settings.gradle .
+COPY src/ ./src/
```

Comment:
not a necessary one but if we are using it later in our project we need to add Gradle caching step for faster build process

Reply:
Do you have any references for that? I did notice when using `gradle` in a container, if you're going to run multiple commands, you want to start up the database and `docker compose run app bash` so that you can leverage the Gradle daemon to speed up subsequent builds.

Reply:
By caching the Gradle dependencies in a Docker volume, we can avoid downloading them repeatedly, which is especially useful in a CI/CD pipeline or when working across multiple projects. This way, the Gradle daemon can leverage both the cached dependencies and its in-memory state to make the build process even faster.
https://stackoverflow.com/questions/25873971/docker-cache-gradle-dependencies
https://docs.gradle.org/current/userguide/build_cache.html

Reply:
Interesting, thanks! I don't think I've explicitly enable the build cache, but I am getting `build` output. I have a volume mapping in `app` so that it appears on the local filesystem (you can then see test result files, for example), and I thought it would be re-used by subsequent containers. I'll see if enabling the build cache does anything for performance.

Reply:
Ah, looks like build cache is stored at `.gradle`.

Reply:
Okay, see this commit. https://github.com/ssciolla/dor-replica-experiment/pull/18/commits/51b1bb390358455f7bfffe4c57e786ec027de092 Seems like there are multiple caches (build and dependency) that are stored in `GRADLE_USER_HOME`. I set that and then mapped a volume for `caches`.